src/jltypes.c

```diff
@@ -3509,7 +3509,7 @@ void jl_init_types(void)
     jl_array_type->pointerfree = 0;
     jl_array_type->ninitialized = 0;
 
-    jl_array_any_type =
+    jl_array_ptr_type =
```

Comment:
Sorry, I'm not familiar with this code. I just fixed it.

I also changed `vector_ptr` to `array_ptr_1d` to be more consistent with existing functions. It's more verbose, but we should rename them all at once if we want shorter identifiers.